UnitWorld_shared/shared/communication/messages/MessageType.h

```diff
@@ -5,6 +5,7 @@ namespace uw
     enum MessageType
     {
         InvalidMessageType,
-        CompleteGameStateMessageType
+        CompleteGameStateMessageType,
+        MoveMobileUnitsToPositionMessageType
```

Comment:
What feels odd with this `enum` is that we are not statically shielded against a client sending a `CompleteGameStateMessageType` or the server sending a `MoveMobileUnitsToPositionMessageType`. To do so, we should split `MessageType`s into 2 categories: `ServerMessageType` and `ClientMessageType`, which would be sent by the server and by a client, respectively.

Reply:
Not sure if we can do this in c++. Nevertheless, this *may* allow to remove the need for casting (not sure either thought), and for sure this would prevent stupid mistakes from happening.

Reply:
Well sure enough we could separate this enum in two, but the the parsing logic behind the messages would also need to be duplicated, or at least to get more complex. I'm really thinking about letting this enum as one because of the work required for the very little profit it would induce. IMO

Reply:
k then